micromanager_gui/_core_widgets/_presets_widget.py

```diff
@@ -0,0 +1,144 @@
+from typing import Optional, Tuple
+
+from pymmcore_plus import DeviceType
+from qtpy.QtWidgets import QComboBox, QHBoxLayout, QWidget
+from superqt.utils import signals_blocked
+
+from .._core import get_core_singleton
+from .._util import get_dev_prop, set_wdg_color
+
+
+class PresetsWidget(QWidget):
+    """Create a QCombobox Widget containing the presets of the specified group"""
+
+    def __init__(
+        self,
+        group: str,
+        text_color: str = "black",
+        parent: Optional[QWidget] = None,
+    ) -> None:
+
+        super().__init__(parent)
+
+        self._mmc = get_core_singleton()
+
+        self._group = group
+        self.text_color = text_color
+
+        if self._group not in self._mmc.getAvailableConfigGroups():
+            raise ValueError(f"{self._group} group does not exist.")
+
+        self._presets = list(self._mmc.getAvailableConfigs(self._group))
+
+        if not self._presets:
+            raise ValueError(f"{self._group} group does not have presets.")
+
+        self.dev_prop = get_dev_prop(self._group, self._presets[0])
+
+        self._check_if_presets_have_same_props()
+
+        self._combo = QComboBox()
+        self._combo.addItems(self._presets)
+
+        self.setLayout(QHBoxLayout())
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(self._combo)
+        set_wdg_color(self.text_color, self._combo)
+        self._combo.currentTextChanged.connect(self._on_combo_changed)
+        self._combo.textActivated.connect(self._on_text_activate)
+
+        self._mmc.events.configSet.connect(self._on_cfg_set)
+        self._mmc.events.systemConfigurationLoaded.connect(self.refresh)
+        self._mmc.events.propertyChanged.connect(self._on_property_changed)
+        # TODO: add connections once we will implement
+        # 'deleteGroup'/'deletePreset signals
+
+        self.destroyed.connect(self.disconnect)
+
+    def _check_if_presets_have_same_props(self):
+        for idx, preset in enumerate(self._presets):
+            if idx == 0:
+                continue
+            device_property = get_dev_prop(self._group, preset)
+            if len(device_property) != len(self.dev_prop):
+                raise ValueError(f"{self._presets} must have the same properties.")
```

Comment:
Can we turn this into a warning instead. Explaining that it will may not update as expected. Alternatively I'm not even sure we need to error/warn here. All we need to do is expand `dev_prop` to make it encompass all possible devices. You've already made `_set_if_props_match_preset` robust enough that it should handle that case :)

Reply:
My prefernce is:
1. not needing this (see below comment for how)
2. warning
3. error

Reply:
@ianhi but if we don't use `_set_if_props_match_preset` to warn the user that the presets are not correct, then what do we need it for?
The change in the `get_dev_prop` method you suggested below is only returning a list, where are we using it? I'm not sure I understand what you mean

Reply:
I'm sorry I'm not being more clear. It's totally possible that (like many times before) I'm actually missing something and asking you to implement something that isn't possible.... I'll try to mess around tonight and generate the test case I'm thinking of (and in process decide if what I'm thinking is on the right track)